drivers/console/Kconfig

```diff
@@ -189,6 +200,35 @@ config UART_PIPE_ON_DEV_NAME
 	  for pipe UART.
 endif
 
+config UART_MCUMGR
+	bool
+	prompt "Enable mcumgr UART driver"
+	select UART_INTERRUPT_DRIVEN
+	default n
+	help
+	  Enable the mcumgr UART driver. This driver allows the application to
+	  communicate over UART using the mcumgr protocol. The driver doesn't
+	  inspect received data (as contrary to console UART driver) and all
+	  aspects of received protocol data are handled by application provided
+	  callback.
+
+if UART_MCUMGR
+if !HAS_DTS
+config UART_MCUMGR_ON_DEV_NAME
+	string "Device Name of UART Device for mcumgr UART"
+	default "UART_0"
+	depends on UART_MCUMGR
+	help
+	  This option specifies the name of UART device to be used
+	  for mcumgr UART.
+endif # !HAS_DTS
+
+config UART_MCUMGR_RX_BUF_SIZE
+	int
+	prompt "Size of receive buffer for mcumgr packets received over UART"
+	default 1024
```

Comment:
A whole kilobyte of RAM? Is that a reasonable default?

For reference, the nrf52832 can at most UART RX 256 bytes over DMA at a time.

Reply:
this is for receive mcumgr protocols data - as i remember in mcuboot a one packet could at be size around half of KB.

Reply:
As @nvlsianpu mentioned, the size of this buffer restricts the MTU for request packets received over UART.  Each mcumgr request is paired with an outgoing response, so the size of this buffer will have a real effect on the speed of an image upgrade operation beyond how fast the server can read requests from its UART.

That said, I agree that 1kB is a lot. I am happy to reduce the default if you feel it is excessive.

Reply:
For now, I would suggest reducing the buffer size to 128 / 256 bytes or so.

I am sure we can find some way to increase the speed in the future. E.g. don't pair each packet with a response,
e.g. by increasing the RX_BUF_SIZE to the image size, use a small RAM buffer and rely on UART's flow control and error detection capability.